### DISCUSSION TITLE
Sharpened B and E notes In a measure

### DISCUSSION THREAD

**Discussion by mxAlexG** [CoderLine/alphaTab#951] [Q&A] [answered]
Hi, 

I noticed when B or E notes get sharpened in a measure while also not being part of the key signature, their subsequent occurrences in the measure get raised a half step higher while being assigned the neutral sign. This is a correct result, but it doesn't follow the standard/tradition. Can this be turned off or easily altered?

This is an example.
![image](https://user-images.githubusercontent.com/105655048/186394952-0291d726-08f9-412f-8fed-51924c12f82c.png)

Thanks and regards...

**Comment by mxAlexG** [comment]
This is how it looks when rendered with Musecore's player.
![image](https://user-images.githubusercontent.com/105655048/186396846-0ff76ec8-3f78-44e7-8309-13dd76af651e.png)

**Comment by Danielku15** [comment, marked as answer]
Do you have an example **file** showing this behavior? From pure screenshots it is hard to judge why alphaTab is placing the accidentals like this. Generally the accidental placement is a core part of the music engraving algorithm. Most of it is done fully automatic and cannot be changed. On individual note level there can be some annotations set which might influence some aspects of the accidental handling. e.g. in Guitar Pro files there is also a way to enforce maybe accidentals over placing a note on a different line and we try to respect those annotations.

Notice how Guitar Pro has the same not heights (according to tabs) on multiple notes, but the standard music notation is displaying it with a different note head position and accidentals: 
![image](https://user-images.githubusercontent.com/674916/187026173-a511106c-1cc9-4bc3-9bb1-ba760e7a5662.png)

and the same file in alphaTab: 
![image](https://user-images.githubusercontent.com/674916/187026294-da9ed468-006d-4e31-bfb8-4c685d2e5c5e.png)

**Reply by Danielku15** [reply]
I also rebuilt your example in Guitar Pro (showing also tabs to indicate the underlying note values a bit more visible) and while technically both tracks have the same notes, they are once shown without an accidental and once with a natural sign. 

![image](https://user-images.githubusercontent.com/674916/187026503-4809c502-70e9-43f8-b9e0-0b3cf9d430a8.png)

The difference is the in the `Note.AccidentalMode` https://github.com/CoderLine/alphaTab/blob/43cbd1f78bf16a4d8a8d086592fd9f278a23d14a/src/model/Note.ts#L453

![image](https://user-images.githubusercontent.com/674916/187026569-d0eea5dd-404e-480f-83a3-bcc279878aa9.png)

It is set to `Default` on the lower variant having the accidental and it is set to `ForceSharp` for the upper variant. The algorithm to calculate the line and accidental to place the note has quite some special cases. A different algorithm and tweaking is technically possible but currently not supported. 

https://github.com/CoderLine/alphaTab/blob/43cbd1f78bf16a4d8a8d086592fd9f278a23d14a/src/rendering/utils/AccidentalHelper.ts#L184
